lib/systems/default.nix

```diff
@@ -1,9 +1,59 @@
-{ lib }:
-  let inherit (lib.attrsets) mapAttrs; in
+{ lib
+}:
+let
+ inherit (lib.attrsets) mapAttrs;
+
+ selectEmulator = platform: pkgs:
+   let
+     qemu-user = pkgs.qemu.override {
+       smartcardSupport = false;
+       spiceSupport = false;
+       openGLSupport = false;
+       virglSupport = false;
+       vncSupport = false;
+       gtkSupport = false;
+       sdlSupport = false;
+       pulseSupport = false;
+       smbdSupport = false;
+       seccompSupport = false;
+       enableDocs = false;
+       hostCpuTargets = [ "${platform.qemuArch}-linux-user" ];
+     };
+     wine = (pkgs.winePackagesFor "wine${toString platform.parsed.cpu.bits}").minimal;
+   in
+     if lib.systems.canExecute pkgs.stdenv.hostPlatform platform
+     then "${pkgs.runtimeShell} -c '\"$@\"' --"
+     else if platform.isWindows
+     then "${wine}/bin/wine${lib.optionalString (platform.parsed.cpu.bits == 64) "64"}"
+     else if platform.isLinux && pkgs.stdenv.hostPlatform.isLinux && platform.qemuArch != null
+     then "${qemu-user}/bin/qemu-${platform.qemuArch}"
+     else if platform.isWasi
+     then "${pkgs.wasmtime}/bin/wasmtime"
+     else if platform.isMmix
+     then "${pkgs.mmixware}/bin/mmix"
+     else null;
 
-rec {
-  doubles = import ./doubles.nix { inherit lib; };
   parse = import ./parse.nix { inherit lib; };
+
+  canExecute = machinePlatform: binaryPlatform:
+    machinePlatform.isAndroid == binaryPlatform.isAndroid &&
+    parse.isCompatible machinePlatform.parsed.cpu binaryPlatform.parsed.cpu
+    && machinePlatform.parsed.kernel == binaryPlatform.parsed.kernel;
+
+  emulatorAvailable = platform: pkgs: (selectEmulator platform pkgs) != null;
+
+  emulator = platform: pkgs:
+    if (emulatorAvailable platform pkgs)
+    then selectEmulator platform pkgs
+    else throw "Don't know how to run ${platform.config} executables.";
+
+ canExecutePlaceholder        = self: plat: canExecute self plat;
+ emulatorAvailablePlaceholder = self: pkgs: emulatorAvailable self pkgs;
+ emulatorPlaceholder          = self: pkgs: emulator self pkgs;
```

Comment:
And this is now unused.

Reply:
See https://github.com/NixOS/nixpkgs/pull/238331#discussion_r1382315387